lib/connection/pool.js

```diff
@@ -83,9 +83,9 @@ var Pool = function(options) {
   this.executing = false;
   // Unref pool
   this.unreference = false;
-  // Set the monitoring connection timeout
+  // Set the monitoring socket timeout
   this.monitoringSocketTimeout = typeof options.monitoringSocketTimeout == 'number'
-    ? options.monitoringSocketTimeout : options.connectionTimeout;
+    ? options.monitoringSocketTimeout : options.socketTimeout;
```

Comment:
should still be connectionTimeout as you want the monitoring process for a replicaset to timeout as fast as possible if there is a hanging socket write so it can retry.